tests/e2e/testsLibrary/ProjectAndFileTests.ts

```diff
@@ -0,0 +1,33 @@
+/*********************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import { CLASSES, Ide, ProjectTree, TestConstants, Editor } from '..';
+import { e2eContainer } from '../inversify.config';
+
+const ide: Ide = e2eContainer.get(CLASSES.Ide);
+const projectTree: ProjectTree = e2eContainer.get(CLASSES.ProjectTree);
+const namespace: string = TestConstants.TS_SELENIUM_USERNAME;
+const editor: Editor = e2eContainer.get(CLASSES.Editor);
+
+export function waitWorkspaceReadiness(workspaceName : string, sampleName : string, folder: string) {
```

Comment:
On my opinion ```waitWorkspaceReadiness``` method doesn't relate to ```ProjectsAndFileTests```
On my opinion it should be moved to the right place, or class should be renamed and reflect its content, or method should be deleted.

Reply:
Besides ```waitWorkspaceReadiness``` name doesn't reflect a functionality of the method,
Because in this method you wait workspace readiness, wait ide and wait until project imported

Reply:
If we'll look at the methods used in *.spec.ts as a BDD definition, the `waitWOrksapceReadiness` makes more sense to me, because the user just wants to wait until the workspace is ready. And he doesn't care about the way it is done - if the test is checking project. So maybe this method should be moved to `WorkspaceHandlingTests` class?

Reply:
On my opinion workspace readiness is a workspace starting and ide downloading. It doesn't include project importing or something else.

Reply:
I think this is a discussion about how much abstract we want the content of specfile to be.
Right now, after reading your discussion I see two ways to proceed and I'm find with either of them:
1) Have more granularity - In this case I believe we should split the method to two methods -> First will be waiting for "workspace/IDE readiness" (meaning IDE is loaded and "user" can start interacting with it) and second will be waiting for project to be imported.
2) Have less granularity - agree, that workspace is "ready", when IDE is loaded **and** project is imported. This way we could implement this in one function/test and call it "waitForWorkspaceReadiness". I know we are hiding some functionality and to overcome that, we could properly document (in form of javadoc) the function (in `ProjectAndFileTests.ts`) 

I'm slightly leaning towards solution `2)`, but even `1)` is OK for me.